File: results/dnd-rest-benchmark/lifecycle-runs/20260730T150315Z_pi_kimi-k2p7-code_typescript-node/src/routes/storage.ts
import { ServerResponse } from 'node:http';
import { resetDatabase, SCHEMA_VERSION, isInitialized } from '../db.js';
import { sendJSON } from '../http-utils.js';

export function storageStatus(res: ServerResponse): void {
  sendJSON(res, 200, {
    driver: 'sqlite',
    schema_version: SCHEMA_VERSION,
    initialized: isInitialized(),
  });
}

export function storageReset(res: ServerResponse): void {
  resetDatabase();
  sendJSON(res, 200, { ok: true, schema_version: SCHEMA_VERSION });
}

File: results/dnd-rest-benchmark/lifecycle-runs/20260731T060905Z_claude_claude-sonnet-5_python-django/dndsite/views/storage.py
"""Schema status/reset introspection endpoints."""

from django.http import JsonResponse
from django.views.decorators.csrf import csrf_exempt

from dndsite import db


def storage_status(request):
    if request.method != "GET":
        return JsonResponse({"error": "method not allowed"}, status=405)
    return JsonResponse(
        {
            "driver": "sqlite",
            "schema_version": db.SCHEMA_VERSION,
            "initialized": db.is_initialized(),
        }
    )


@csrf_exempt
def storage_reset(request):
    if request.method != "POST":
        return JsonResponse({"error": "method not allowed"}, status=405)
    db.reset_schema()
    return JsonResponse({"ok": True, "schema_version": db.SCHEMA_VERSION})
